File: CarHomeWidget/src/com/anod/car/home/skin/BBBIconProcessor.java
package com.anod.car.home.skin;

import android.graphics.Bitmap;

public class BBBIconProcessor implements IconProcessor {

	@Override
	public Bitmap process(Bitmap icon) {
		int x = (int)(icon.getWidth() * 0.3f);
		int w = icon.getWidth() - x;
		int y = (int)(icon.getHeight() * 0.3f);
		int h = icon.getHeight() - y;

		Bitmap quarter = Bitmap.createBitmap(icon, x, 0, w, h);
		return quarter;
	}

}

File: skins/src/main/java/info/anodsplace/carwidget/skin/icon/BBBIconProcessor.kt
package info.anodsplace.carwidget.skin.icon

import android.graphics.Bitmap
import info.anodsplace.carwidget.content.graphics.IconProcessor

class BBBIconProcessor : IconProcessor {

    override fun process(icon: Bitmap): Bitmap {
        val x = (icon.width * sizeDiff).toInt()
        val w = icon.width - x
        val y = (icon.height * sizeDiff).toInt()
        val h = icon.height - y

        return Bitmap.createBitmap(icon, x, 0, w, h)
    }

    override val sizeDiff = .3f
    override val id = "bbb"

}